Candidate for buildword solution

bediger4000 opened the pull request:
Hello. I think this PR satisfies your buildword challenge.

I think that your first test is incorrect: 

    {"buildword", []string{"buil", "dwor", "bu", "uild", "wo", "rd"}, 4},

I believe there's no combination of those 6 fragments that will get you to "buildword" - either "bu" should be "be", or "uild" should be "ild"

plutov commented:
Thanks, I will change test in master.

plutov commented:
Also I will check all PRs on the end of the week. Please pull new master and check it again. Also please remove unused comments.

bediger4000 commented:
I removed commented-out code, changed a few variable names for clarity, removed a struct member that didn't get accessed outside of commented-out-code. Hope this meets your ideal of a solution, it now passes all of your test cases.
